doc/adrs/161_add_support_for_actions_in_tree_item_menu.adoc

```diff
@@ -0,0 +1,180 @@
+= ADR-161 Add support for actions in tree item menu
+
+== Context
+
+Currently, actions in tree item context menu are hard coded in the front end and one cannot add or remove an action without recompiling the front end code.
+
+== Decision
+
+We need to provide a general way to handle actions in tree item contextual menu that does not require front end code modification.
+
+=== GraphQL
+
+First we need to define a new query that returns the list of tree item contextual menu action.
+
+```graphql
+type TreeDescription implements RepresentationDescription {
+  ...
+  contextMenu(treeItemId: ID!): [TreeItemContextActionMetadata!]!
+  ...
+}
+
+interface TreeItemContextActionMetadata {
+  id: ID!
+  label: String!
+  iconURL: [String!]!
+}
+
+type TreeItemSimpleContextActionMetadata implements TreeItemContextActionMetadata {
+  id: ID!
+  label: String!
+  iconURL: [String!]!
+}
+
+type TreeItemFetchContextActionMetadata implements TreeItemContextActionMetadata {
+  id: ID!
+  label: String!
+  iconURL: [String!]!
+  urlToFetch : String!
+  fetchKind : FetchKind!
+}
+
+enum FetchKind {
+  DOWNLOAD
+  OPEN
```

Comment:
What would open do? open in place? open in a new tab? and why one instead of the other?

Reply:
It seems that is required for fetch menu entries to be able to perform a download of a given file or open a page inside another tab